async-nats/src/jetstream/consumer.rs

```diff
@@ -15,6 +15,99 @@ use std::time::Duration;
 use serde::{Deserialize, Serialize};
 use time::serde::rfc3339;
 
+use crate::Error;
+
+pub trait IntoConsumerConfig {
+    fn into_consumer(self) -> ConsumerConfig;
```

Comment:
Not sure if this should be ref self, ala `ToSocketAddrs` or an into trait like now.
But name should at-least end with `consumer_config`.

```suggestion
    fn into_consumer_config(self) -> ConsumerConfig;
```